docs/DeveloperGuide.md

```diff
@@ -781,6 +781,89 @@ The following UML sequence diagram illustrates how the FindTags operations works
 
 --------------------------------------------------------------------------------------------------------------------
 
+### 3.12 Sorting patients by specified attributes
+
+#### Introduction
+
+The `SortCommand` class is responsible for sorting the patients by the specified attribute.
+
+#### Specifications
+
+* `SortCommand` takes in one zero attributes to sort the patient list.
+* `SortCommand` will update the patient list with the sorted patient list and the display the sorted patient list.
+* If no attribute is specified, the default sorting method is by patient name.
+* SortCommand only can sort by patient's name or patient's preferred name.
+* If multiple patients have the same name, the original order (with respect to the affected patients only) will be
+preserved for the affected patients.
+
+The activity diagram below outlines the steps involved when a user initiates a Delete Event command.
+<puml src="diagrams/SortActivityDiagram.puml" alt="SortActivityDiagram" />
+
+#### Example Usage Scenario
+
+Given below is an example usage scenario and how the group creation mechanism behaves at each step.
+
+Step 1: The user accesses the PatientSync application.
+
+Step 2: The user executes `sort p` to sort patient list by patient's preferred name.
```

Comment:
Would it be better to also include example of sorting patient list by patient's name here as well?

Reply:
Sorting by patient name follows a similar flow of events, thus I feel that showing one example should be sufficient!